impeller/renderer/backend/gles/sampler_gles.cc

```diff
@@ -16,19 +16,20 @@ SamplerGLES::SamplerGLES(SamplerDescriptor desc) : Sampler(std::move(desc)) {}
 
 SamplerGLES::~SamplerGLES() = default;
 
+// TODO(https://github.com/flutter/flutter/issues/148253): `MipFilter::kNearest`
+// is the default value for mip filters, as it cooresponds with the framework's
+// `FilterQuality.low`. If we ever change the default filter quality, we should
+// update this function to match.
 static GLint ToParam(MinMagFilter minmag_filter,
-                     std::optional<MipFilter> mip_filter = std::nullopt) {
-  if (!mip_filter.has_value()) {
-    switch (minmag_filter) {
-      case MinMagFilter::kNearest:
-        return GL_NEAREST;
-      case MinMagFilter::kLinear:
-        return GL_LINEAR;
-    }
-    FML_UNREACHABLE();
-  }
-
-  switch (mip_filter.value()) {
+                     MipFilter mip_filter = MipFilter::kNearest) {
+  switch (mip_filter) {
```

Comment:
@chinmaygarde  to confirm, I think I remember this being weird because I had to fake support no mips.

Reply:
I suspect we still need to treat an unspecified param as kBase here

Reply:
PTAL again, made some adjustments

Reply:
looks good to me